pkg/kube/container/container_util.go

```diff
@@ -0,0 +1,14 @@
+package container
+
+import corev1 "k8s.io/api/core/v1"
+
+// GetByName returns a container with the given name from the slice of containers.
+// nil is returned if the container does not exist.
+func GetByName(name string, containers []corev1.Container) *corev1.Container {
```

Comment:
allows us to not worry about the order of the containers, we can always fetch the container by name rather than index.